util/Setup/Program.cs

```diff
@@ -218,26 +226,61 @@ private static bool ValidateInstallation()
         }
         else
         {
-            installationKey = Helpers.ReadInput("Enter your installation key");
+            var prompt = "Enter your installation key";
+            installationKey = Helpers.ReadInput(prompt);
+            while (installationKey == string.Empty)
+            {
+                Helpers.WriteError("Invalid input for installation key. Please try again.");
+                installationKey = Helpers.ReadInput(prompt);
+            }
+        }
+
+        if (_context.Parameters.ContainsKey("cloud-region"))
+        {
+            Enum.TryParse(_context.Parameters["cloud-region"], out cloudRegion);
```

Comment:
This will take something like `blah` and treat it like `CloudRegion.US`, is that what we want?

Reply:
Yes.  We currently don't have a use for running the setup container with `cloud-region` specified, but for backwards compatibility I believe the default should be `US`.